beacon-chain/core/helpers/validators.go

```diff
@@ -153,6 +168,12 @@ func BeaconProposerIndex(state *pb.BeaconState) (uint64, error) {
 		return 0, errors.Wrap(err, "could not get active indices")
 	}
 
+	if featureconfig.Get().EnableProposerIndexCache {
+		if err := UpdateCommitteeCache(state, CurrentEpoch(state)); err != nil {
```

Comment:
if we have the flag, this will update the committee cache everytime `BeaconProposerIndex` is called. That will be very expensive and wasteful. Maybe we can find a better way to cache the proposer indexes for this ?

Reply:
No it wouldn't, It'll only update the cache once at the beginning of every epoch. I tested run time with latest cluster